### DISCUSSION TITLE
How to create JDBC data source using SQL Server Driver

### DISCUSSION THREAD

**Discussion by stone2dbone** [eclipse-birt/birt#1098] [Q&A]
I cannot create a JDBC data source using the BIRT Report Design plugin and Eclipse IDE for Enterpise Java and Web Developers Version:2022-09 (4.25.0) Build id 20220908-1902.

Using JDBC Data Source:

![Annotation 2022-10-27 141729](https://user-images.githubusercontent.com/19313052/198367813-31a06951-5716-4f57-a9e3-c68e16ef6198.jpg)

I have attempted each of the following URLs:

![Annotation 2022-10-27 141622](https://user-images.githubusercontent.com/19313052/198367957-7248cd91-c4fa-400e-bda7-54d0b6e2c59b.jpg)

![Annotation 2022-10-27 141445](https://user-images.githubusercontent.com/19313052/198368069-49ba157a-76ff-4a8c-a1c3-64e339e273f4.jpg)

I repeatedly get the following error:

![Annotation 2022-10-27 141423](https://user-images.githubusercontent.com/19313052/198368236-cff92a47-38cf-4f18-8aa9-aa4c8d2a8c6f.jpg)

Using the JDBC Database Connection for Query Builder, with the SQL Server Connection Profile Type, 

![Annotation 2022-10-27 142157](https://user-images.githubusercontent.com/19313052/198369370-9779d543-9b02-4c14-b437-7cd465e49cc7.jpg)

I get the following error after modifying the driver properties:

![Annotation 2022-10-27 142315](https://user-images.githubusercontent.com/19313052/198369584-0b2369f1-6405-40b9-9267-5b484077b3b8.jpg)

If I back out and then start over to create a new data source using the SQL Server Connection Profile Type I previously modified, I get this error:

![Annotation 2022-10-27 142510](https://user-images.githubusercontent.com/19313052/198370099-ae650231-1147-4c35-921b-ba23744ed8b7.jpg)

I am using mssql-jdbc-6.1.0.jre8.jar. I have unsuccessfully tried to use 7.0.0.jre8 and 11.2.0.jre8 as well.

(NOTE: I just installed Eclipse Report Designer package, the portable Eclipse installation with the BIRT plugin pre-installed, and was able to connect without a problem! I'd prefer not to have to switch between IDEs, so if someone can offer a solution to my original problem, it would still be greatly appreciated.)

**Comment by speckyspooky** [comment]
Hello,

I have on my workstation-laptop an MS SQL Server (SQL-Express) and I tested 2 JDBC drivers for the connection and both works fine.
The first one is the oldest JDBC-driver of MSSQL the JTDS-driver and the second is the original Microsoft JDBC-driver "com.microsoft.sqlserver.jdbc.SQLServerDriver ( v10.2)".

I have located both drivers on an separated folder and added both drivers with the according jar-files on the driver manager (Manage Drivers).

Attached I added 2 screens of the "Manage Driver"-overview and my connection configuration.

Perhaps the screens will help to get some information to solve your problem.

***Driver administration:***

![ManageDriver-MSSQL-00](https://user-images.githubusercontent.com/41593722/198833813-07c500c6-9ddd-4602-ae82-96c7cb47287e.png)

***Connection properties***
![ManageDriver-MSSQL-01](https://user-images.githubusercontent.com/41593722/198833824-4239380e-d524-43de-8e91-78bf1e350637.png)

**Reply by stone2dbone** [reply]
Thanks, but the only difference I see is that you've added "trustServerCertificate=true". Using the BIRT Report Design plugin I'm still not able to connect after adding that parameter to the URL.

**Reply by speckyspooky** [reply]
The issue sames to me a local issue becuase the connection is possible.
Can you please retry the connection by removing the drivers -> eclipse restart, add the driver again -> eclipse restart, create your connection and send the error message of the Java stacktrace. Perhaps we can see together some hint.

I don't use connection profiles. I use only the direct connections.

**Reply by EAwesomeK** [reply]
> Hello,
> 
> I have on my workstation-laptop an MS SQL Server (SQL-Express) and I tested 2 JDBC drivers for the connection and both works fine. The first one is the oldest JDBC-driver of MSSQL the JTDS-driver and the second is the original Microsoft JDBC-driver "com.microsoft.sqlserver.jdbc.SQLServerDriver ( v10.2)".
> 
> I have located both drivers on an separated folder and added both drivers with the according jar-files on the driver manager (Manage Drivers).
> 
> Attached I added 2 screens of the "Manage Driver"-overview and my connection configuration.
> 
> Perhaps the screens will help to get some information to solve your problem.
> 
> _**Driver administration:**_
> 
> ![ManageDriver-MSSQL-00](https://user-images.githubusercontent.com/41593722/198833813-07c500c6-9ddd-4602-ae82-96c7cb47287e.png)
> 
> _**Connection properties**_ ![ManageDriver-MSSQL-01](https://user-images.githubusercontent.com/41593722/198833824-4239380e-d524-43de-8e91-78bf1e350637.png)

How do you get 2 SQL drivers to show up in the driver class drop down menu?  I can have 2 drivers but the newest one is the only 1 that shows up in the drop down list.
